### DISCUSSION TITLE
Microservices with Prisma

### DISCUSSION THREAD

**Discussion by muhammad-Osman** [prisma/prisma#15566] [Q&A]
### Bug description

I'm working on a project right now that has multiple TypeScript microservices performing operations on the same database. In each microservice, we are using the Prisma client to perform database operations. The issue I'm experiencing is we need to duplicate our schema.prisma file in each microservice and generate the Prisma client for each service. Is there a way to manage our database in a separate project while sharing the generated client between the microservices without having duplicate schema.prisma in each project?

### Prisma information

```prisma
// Add your schema.prisma
```

```ts
// Add your code using Prisma Client
```

### Environment & setup

- OS: 
- Database: 
- Node.js version: 

### Prisma Version

```
4.3.0
```

**Comment by nurul3101** [comment]
Hey @muhammad-Osman 👋 

You can create a private npm package which contains your schema file and PrismaClient and you can import them across all your projects. Here's an [article](https://isidoro-ferreiro.medium.com/share-your-prisma-client-across-projects-44d1c7aca6fd) written by one of the community members on how they shared PrismaClient across multiple projects.

**Reply by vladi-stevanovic** [reply]
👋 Hello @muhammad-Osman - did you perhaps have a chance to check Nurul's recommendation? Let us know if you still have any questions, or feel free to "Mark as answer" if this discussion is now solved!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
